landingai/predict.py

```diff
@@ -193,228 +163,497 @@ def predict(
+    ) -> List[SegmentationPrediction]:
+        """Extract Segmentation prediction result from response
+
+        Parameters
+        ----------
+        response: Response from the Edge prediction endpoint.
+
+        """
+        encoded_predictions = response["predictions"]["bitmaps"]
+        encoding_map = response["predictions"]["encoding"]["options"]["map"]
+        mask_shape = (
+            response["predictions"]["imageHeight"],
+            response["predictions"]["imageWidth"],
+        )
+        return [
+            SegmentationPrediction(
+                id=id,
+                label_name=bitmap_info["labelName"],
+                label_index=bitmap_info["labelIndex"],
+                score=bitmap_info["score"],
+                encoded_mask=bitmap_info["bitmap"],
+                encoding_map=encoding_map,
+                mask_shape=mask_shape,
+            )
+            for id, bitmap_info in encoded_predictions.items()
+        ]
+
+    @staticmethod
+    def extract_prediction(response: Dict[str, Any]) -> List[Prediction]:
+        response_type = response.get("type")
+        if response_type is None:
+            response_type = response["type"]
+        predictions: List[Prediction]
+        if response_type == "ObjectDetectionPrediction":
+            predictions = _EdgeExtractor._extract_edge_od_prediction(response)  # type: ignore
+        elif response_type == "SegmentationPrediction":
+            predictions = _EdgeExtractor._extract_edge_seg_prediction(response)  # type: ignore
+        elif response_type == "ClassificationPrediction":
+            predictions = _EdgeExtractor._extract_edge_class_prediction(response)  # type: ignore
+        else:
+            raise NotImplementedError(
+                f"{response_type} is not implemented in EdgeExtractor"
+            )
+        return predictions
+
+
+def _create_session(url: str, num_retry: int, headers: Dict[str, str]) -> Session:
+    """Create a requests session with retry"""
+    session = Session()
+    retries = Retry(
+        # TODO: make them configurable
+        total=num_retry,
+        backoff_factor=3,
+        raise_on_redirect=True,
+        raise_on_status=False,  # We are already raising exceptions during backend invocations
+        allowed_methods=["GET", "POST", "PUT"],
+        status_forcelist=[
+            # 408 Request Timeout , 413 Content Too Large, 500 Internal Server Error
+            429,  # Too Many Requests  (ie. rate limiter)
+            502,  # Bad Gateway
+            503,  # Service Unavailable
+            504,  # Gateway Timeout
+        ],
+    )
+    session.mount(
+        url, HTTPAdapter(max_retries=retries)
+    )  # Since POST is not idempotent we will ony retry on the this specific API
+    session.headers.update(headers)
+    return session
+
+
+def _do_inference(
+    session: Session,
+    endpoint_url: str,
+    files: List[Any],
+    payload: Dict[str, Any],
+    extractorClass: Type[_Extractor],
```

Comment:
nit: should use snake case naming, i.e. `extractor_class`

Reply:
changed